test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs

```diff
@@ -443,4 +456,33 @@ public void ShouldIgnoreStaticConstructorAndDiagnostic()
             .HaveDiagnostic(DiagnosticDescriptors.CouldNotCreateMapping)
             .HaveAssertedAllDiagnostics();
     }
+
+    [Fact]
+    public void ShouldFlattedNestedToTargetRootMapping()
```

Comment:
Since this is a complete new feature, an integration test should be added to ensure this works with all supported Roslyn versions.